app/components/SingleProduct.jsx

```diff
@@ -1,17 +1,118 @@
 import React from 'react'
-
 import { login } from 'APP/app/reducers/auth'
 import { connect } from 'react-redux'
 
-const SingleProduct = (props) => (
-  <div>
-    <h2>Product name</h2>
-    <h3>Rest of the product description</h3>
-  </div>
+// *------------------- COMPONENT -----------------*//
+
+class SingleProduct extends React.Component {
+
+  constructor(props) {
+    super(props)
+    this.state = {
+      color: '',
+      size: '',
+      quantity: 0
+    }
+    this.setColor = this.setColor.bide(this)
+    this.setSize = this.setSize.bide(this)
+    this.setQuantity = this.setQuantity.bide(this)
+  }
+
+  setColor(val){
+    this.setState({ color: val })
+  }
+
+  setSize(val){
+    this.setState({ size: val })
+  }
+
+  setQuantity(val){
+    this.setState({ quantity: val})
+  }
+
+  render() {
+    const product = this.props.selectedProduct
+    const reviews = this.props.reviews
+
+    return (
+      <div>
+        <div className = "col-md-6">
+          <div className="carousel carousel-slider">
+            <a className="carousel-item" href="#one!">
+              {
+                product && product.images.map(image => (
+                  <img src={ image } />
+                ))
+              }
+            </a>
+          </div>
+        </div>
+
+        <div className="col-md-6">
+          <form onSubmit = { this.props.handleSubmit }>
+            <div className="container">
+              <h2 className="panel-title large-font">{ product.name }</h2>
+
+              <p>Price: { product.price }</p>
+              <div className = "tb">{ product.description }</div>
+              <a className='dropdown-button btn' href='#' data-activates='dropdown1'>Color</a>
+              <ul id='dropdown1' className='dropdown-content' >
+                {
+                  product.color && product.color.map(color => (
+                    <li><a onChange = {(e) => this.setColor(e.target.value) } >{ color }</a></li>
+                  ))
+                }
+              </ul>
+
+              <a className='dropdown-button btn' href='#' data-activates='dropdown1'>Size</a>
+              <ul id='dropdown1' className='dropdown-content'>
+                {
+                  product.size  && product.size.map(size => (
+                    <li><a onChange = {(e) => this.setSize(e.target.value) } >{ size }</a></li>
+                  ))
+                }
+              </ul>
+
+              <a className = "waves-effect waves-light btn-large"
+                 type="submit"
+                 disabled={ this.state.color === '' || this.state.size === '' || this.state.quantity === 0 } >
+                ADD TO CART </a>
+            </div>
+          </form>
+        </div>
+
+
+        <ul className= "list-group">
+          {
+            reviews && reviews.filter(review => review.productId === product.id)
```

Comment:
I am pretty sure that what you're trying to achieve here will be managed by the database. For example, each product should have only certain reviews associated with it already, so we won't have to filter through ALL of the reviews. 

Maybe I'm wrong though?